Cannot open a browser session for profile with sso

Hi, 

**Thanks for this initiative, nice work.**

I've tried to open the browser (chrome) and I've this issue on below. 

Could you please help me about that ?

 **cat ~/.aws/config**

  _[profile admin-dev]
  sso_start_url = https://test.awsapps.com/start
  sso_region = eu-west-1
  sso_account_id = 1234567890
  sso_role_name = AdministratorAccess
  region = eu-west-1_

**assume -c**

? Please select the profile you would like to assume:  admin-dev
If browser is not opened automatically, please open link:
https://device.sso.eu-west-1.amazonaws.com/?user_code=NNJW-TVSX

Awaiting authentication in the browser...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1024de644]

goroutine 1 [running]:
github.com/99designs/keyring.(*fileKeyring).unlock(0x140003c93b0)
	/Users/runner/go/pkg/mod/github.com/99designs/keyring@v1.2.0/file.go:70 +0xa4
github.com/99designs/keyring.(*fileKeyring).Set(0x140003c93b0, {{0x1400017acc0, 0x20}, {0x14000547b80, 0x527, 0x580}, {0x0, 0x0}, {0x0, 0x0}, ...})
	/Users/runner/go/pkg/mod/github.com/99designs/keyring@v1.2.0/file.go:139 +0x98
github.com/common-fate/granted/pkg/credstore.Store({0x1400017acc0, 0x20}, {0x1027fd200, 0x140003c91a0})
	/Users/runner/work/granted-cli-build/granted-cli-build/granted/pkg/credstore/credstore.go:35 +0xdc
github.com/common-fate/granted/pkg/cfaws.StoreSSOToken({0x1400017acc0, 0x20}, {{0x140001a8000, 0x4e9}, {0xc07e2169a6aefc10, 0x1a3373598536, 0x102ca3240}})
	/Users/runner/work/granted-cli-build/granted-cli-build/granted/pkg/cfaws/secure_token_storage.go:32 +0x64
github.com/common-fate/granted/pkg/cfaws.(*CFSharedConfig).SSOLogin(0x14000264780, {0x102894c70, 0x14000144008})
	/Users/runner/work/granted-cli-build/granted-cli-build/granted/pkg/cfaws/sso.go:61 +0x1ac
github.com/common-fate/granted/pkg/cfaws.(*CFSharedConfig).Assume(0x14000264780, {0x102894c70, 0x14000144008})
	/Users/runner/work/granted-cli-build/granted-cli-build/granted/pkg/cfaws/profiles.go:190 +0x1d8
github.com/common-fate/granted/pkg/assume.AssumeCommand(0x14000143ac0)
	/Users/runner/work/granted-cli-build/granted-cli-build/granted/pkg/assume/assume.go:88 +0x610
github.com/common-fate/granted/pkg/updates.WithUpdateCheck.func1(0x14000143ac0)
	/Users/runner/work/granted-cli-build/granted-cli-build/granted/pkg/updates/updates.go:69 +0xc8
github.com/urfave/cli/v2.(*App).RunContext(0x14000242820, {0x102894c70, 0x14000144008}, {0x14000148000, 0x2, 0x2})
	/Users/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:322 +0x6e8
github.com/urfave/cli/v2.(*App).Run(...)
	/Users/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.main()
	/Users/runner/work/granted-cli-build/granted-cli-build/granted/cmd/assume/main.go:13 +0x60

--- a/pkg/browsers/detect.go
+++ b/pkg/browsers/detect.go
@@ -154,6 +154,7 @@ func HandleBrowserWizard(ctx *cli.Context) error {
 
 //ConfigureBrowserSelection will verify the existance of the browser executable and promot for a path if it cannot be found
 func ConfigureBrowserSelection(browserName string) error {
+	withStdio := survey.WithStdio(os.Stdin, os.Stderr, os.Stderr)
 	browserTitle := strings.Title(strings.ToLower(GetBrowserName(browserName)))
 	// We allow users to configure a custom install path is we cannot detect the installation
 	customBrowserPath := ""
@@ -168,7 +169,7 @@ func ConfigureBrowserSelection(browserName string) error {
 
 			bpIn := survey.Input{Message: fmt.Sprintf("Please enter the full path to your browser installation for %s:", browserTitle)}
 			fmt.Fprintln(os.Stderr)
-			err := testable.AskOne(&bpIn, &customBrowserPath)
+			err := testable.AskOne(&bpIn, &customBrowserPath, withStdio)
 			if err != nil {
 				return err
 			}

--- a/pkg/credstore/credstore.go
+++ b/pkg/credstore/credstore.go
@@ -59,8 +59,7 @@ func openKeyring() (keyring.Keyring, error) {
 	credStorePath := path.Join(grantedFolder, "cred-store")
 
 	ring, err := keyring.Open(keyring.Config{
-		AllowedBackends: []keyring.BackendType{keyring.FileBackend},
-		FileDir:         credStorePath,
+		FileDir: credStorePath,
 		FilePasswordFunc: func(s string) (string, error) {
 			in := survey.Password{Message: s}
 			var out string

--- a/pkg/credstore/credstore.go
+++ b/pkg/credstore/credstore.go
@@ -58,7 +58,7 @@ func openKeyring() (keyring.Keyring, error) {
 	// check if the cred-store file exists in the folder
 	credStorePath := path.Join(grantedFolder, "cred-store")
 
-	ring, err := keyring.Open(keyring.Config{
+	return keyring.Open(keyring.Config{
 		FileDir: credStorePath,
 		FilePasswordFunc: func(s string) (string, error) {
 			in := survey.Password{Message: s}
@@ -69,12 +69,4 @@ func openKeyring() (keyring.Keyring, error) {
 		},
 		ServiceName: "granted",
 	})
-	if err != nil {
-		return nil, err
-	}
-	// Ensure this keyring is correctly opened, there have been some issues with ring being nil with no error
-	if ring == nil {
-		return nil, ErrCouldNotOpenKeyring
-	}
-	return ring, err
 }

--- a/pkg/credstore/credstore.go
+++ b/pkg/credstore/credstore.go
@@ -2,13 +2,17 @@ package credstore
 
 import (
 	"encoding/json"
+	"errors"
 	"os"
 	"path"
 
 	"github.com/99designs/keyring"
+	"github.com/AlecAivazis/survey/v2"
 	"github.com/common-fate/granted/pkg/config"
-	"github.com/common-fate/granted/pkg/debug"
+	"github.com/common-fate/granted/pkg/testable"
 )
+
+var ErrCouldNotOpenKeyring error = errors.New("keyring not opened successfully")
 
 // returns ring.ErrKeyNotFound if not found
 func Retrieve(key string, target interface{}) error {
@@ -26,7 +30,7 @@ func Retrieve(key string, target interface{}) error {
 func Store(key string, payload interface{}) error {
 	ring, err := openKeyring()
 	if err != nil {
-		return nil
+		return err
 	}
 	b, err := json.Marshal(payload)
 	if err != nil {
@@ -41,7 +45,7 @@ func Store(key string, payload interface{}) error {
 func Clear(key string) error {
 	ring, err := openKeyring()
 	if err != nil {
-		return nil
+		return err
 	}
 	return ring.Remove(key)
 }
@@ -51,24 +55,27 @@ func openKeyring() (keyring.Keyring, error) {
 	if err != nil {
 		return nil, err
 	}
-
 	// check if the cred-store file exists in the folder
 	credStorePath := path.Join(grantedFolder, "cred-store")
-	_, err = os.Stat(credStorePath)
-	fileExists := err == nil
 
-	if !fileExists {
-		debug.Fprintf(debug.VerbosityDebug, os.Stderr, "ℹ️  A cred-store file was not found\n")
-		debug.Fprintf(debug.VerbosityDebug, os.Stderr, "Creating cred-store file at %s\n", credStorePath)
-		_, err = os.Create(credStorePath)
-		if err != nil {
-			return nil, err
-		}
-
-	}
-	return keyring.Open(keyring.Config{
-		FileDir:     path.Join(grantedFolder, "cred-store"),
+	ring, err := keyring.Open(keyring.Config{
+		AllowedBackends: []keyring.BackendType{keyring.FileBackend},
+		FileDir:         credStorePath,
+		FilePasswordFunc: func(s string) (string, error) {
+			in := survey.Password{Message: s}
+			var out string
+			withStdio := survey.WithStdio(os.Stdin, os.Stderr, os.Stderr)
+			err := testable.AskOne(&in, &out, withStdio)
+			return out, err
+		},
 		ServiceName: "granted",
 	})
-
+	if err != nil {
+		return nil, err
+	}
+	// Ensure this keyring is correctly opened, there have been some issues with ring being nil with no error
+	if ring == nil {
+		return nil, ErrCouldNotOpenKeyring
+	}
+	return ring, err
 }

